### PR TITLE
[REVIEW] Update fillna()

### PR DESCRIPTION
**Pull request by shwina**
Addresses #1023

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by kkraus14** [comment]
add to whitelist

File: python/cudf/dataframe/columnops.py [hunk 1]
```diff
@@ -333,6 +336,9 @@ def as_column(arbitrary, nan_as_null=True, dtype=None):
         else:
             data = as_column(pa.array(arbitrary, from_pandas=nan_as_null))
 
+    elif isinstance(arbitrary, pd.Timestamp):
+        data = as_column(pa.array([arbitrary]))
```
Comment (kkraus14) [inline python/cudf/dataframe/columnops.py]:
```suggestion
        data = as_column(pa.array([arbitrary], from_pandas=True))
```

**Comment by shwina** [comment]
Thanks! Should it have been `from_pandas=nan_as_null`?

**Comment by kkraus14** [comment]
> Thanks! Should it have been `from_pandas=nan_as_null`?

This actually doesn't work for nulls in general:
```
In [82]: pa.array([pd.Timestamp(np.nan)])                                                                            
Out[82]: 
<pyarrow.lib.TimestampArray object at 0x7f031a880228>
[
  -62135596800000000
]

In [83]: pa.array([pd.Timestamp(np.nan)], from_pandas=True)                                                          
Out[83]: 
<pyarrow.lib.TimestampArray object at 0x7f031a87af98>
[
  -62135596800000000
]
```

If you construct a pandas series from the timestamp it returns null no matter what:
```
In [89]: pa.array(pd.Series([pd.Timestamp(np.nan)]), from_pandas=False)                                              
Out[89]: 
<pyarrow.lib.TimestampArray object at 0x7f031a9beb88>
[
  null
]

In [90]: pa.array(pd.Series([pd.Timestamp(np.nan)]), from_pandas=True)                                               
Out[90]: 
<pyarrow.lib.TimestampArray object at 0x7f031a8b5188>
[
  null
]
```

**Comment by shwina** [comment]
Gotcha. Working on quick addition of `fillna()` and then this should be OK to merge

**Comment by kkraus14** [comment]
> Gotcha. Working on quick addition of `inplace` and then this should be OK to merge

We actually need to address that columnops.as_column of a timestamp issue. I'm writing up a quick workaround hack.

**Comment by kkraus14** [comment]
> > Gotcha. Working on quick addition of `inplace` and then this should be OK to merge
> 
> We actually need to address that columnops.as_column of a timestamp issue. I'm writing up a quick workaround hack.

Wrote a change which always treats NaT as null. There isn't technically a discrete value for datetimes / timestamps since they use an int64 underneath so I think this is fine. If this bites someone they'll submit a bug and we can address then but I'm really not sure what the expected behavior would be.

**Comment by kkraus14** [comment]
rerun tests

**Pull request merged**